### PR TITLE
Add Multiresolution HuBERT as an additional upstream model

### PR DESCRIPTION
**Pull request by ftshijt**
As per discussion in #515 ,this PR adds the multiresolution HuBERT as an additional upstream model.

TODOs:
- docs
- upload converted ckpt to huggingface (as well as related entry in `hubconf.py`)

Reference PR in fairseq:
- https://github.com/facebookresearch/fairseq/pull/5363
- https://arxiv.org/abs/2310.02720

Btw, during the evaluation of vc task, also fixed a minor bug related to new updates in librosa

### CODE REVIEW

**Comment by leo19941227** [comment]
Hi @ftshijt !

Sure, I will review the changes this weekend. Thanks so much!

File: docs/source/tutorial/upstream_collection.rst [hunk 1]
```diff
@@ -973,6 +973,69 @@ wavlablm_ms_40k
...
+
+
+multires_hubert_multilingual_large600k
+--------------------------------------
+
+- Unlabeled Speech: Voxpopuli 100khr
+- K-means extracted from `hubert_base`_
+- Training steps 600k
+
+
```
Comment (leo19941227) [inline docs/source/tutorial/upstream_collection.rst:1038]: For the K-means extraction source, the current document implies that all the models are using the K-means from the 2nd iteration of HuBERT Base, could you help double check if this is true?
For example, this means `multires_hubert_base` is the 3rd iteration of the HuBERT training, which will not be comparable to `hubert_base`.

File: s3prl/downstream/a2o-vc-vcc2020/utils.py [hunk 2]
```diff
@@ -333,7 +333,7 @@ def logmelspc_to_linearspc(lmspc, fs, n_mels, n_fft, fmin=None, fmax=None):
     fmin = 0 if fmin is None else fmin
     fmax = fs / 2 if fmax is None else fmax
     mspc = np.power(10.0, lmspc)
-    mel_basis = librosa.filters.mel(fs, n_fft, n_mels, fmin, fmax)
+    mel_basis = librosa.filters.mel(sr=fs, n_fft=n_fft, n_mels=n_mels, fmin=fmin, fmax=fmax)
```
Comment (leo19941227) [inline s3prl/downstream/a2o-vc-vcc2020/utils.py:336]: Thanks for the fix!

File: s3prl/upstream/multires_hubert/expert.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        super().__init__(**kwargs)
+        model, task_cfg = load_converted_model(ckpt)
+        self.model = model
+        self.task_cfg = task_cfg
+        self.final_factors = []
+
+        self.num_res = self.model.label_nums
+        logger.info("num res: {}".format(self.num_res))
+        # decide ratios
+        if legacy:
```
Comment (leo19941227) [inline s3prl/upstream/multires_hubert/expert.py:42]: There is no entry with `legacy=False`, what does this option mean in this case?

**Review by leo19941227 — COMMENTED** [review]
Thanks @ftshijt !
I have a few comments, please help check it.
Thanks!

File: s3prl/upstream/multires_hubert/expert.py (hunk 3 above)
Comment (ftshijt) [reply s3prl/upstream/multires_hubert/expert.py:42]: My bad, will fix it soon

File: docs/source/tutorial/upstream_collection.rst (hunk 1 above)
Comment (ftshijt) [reply docs/source/tutorial/upstream_collection.rst:1038]: Yeah, the current note is true. All the provided models shall be considered as the 3rd iteration of the HuBERT-training (not directly comparable to hubert_base). In our provided paper, we also conducted the experiments with the "comparable" hubert using the same 3rd iteration k-means.

In that case, do you suggest me submit those models (3rd iteration hubert) to the s3prl as well?

**Comment by ftshijt** [comment]
> Thanks @ftshijt ! I have a few comments, please help check it. Thanks!

Many thanks for the review! I've removed the unused legacy note, but I'm not sure if we would also like to add the 3rd iteration hubert-base as discussed above. But I feel even in that case, I can probably simply add that to the s3prl/hubert huggingface repo and leave it there for people who are interested in a fair comparison between hubert and mr-hubert (let me know if you have any other better ideas~).

File: docs/source/tutorial/upstream_collection.rst (hunk 1 above)
Comment (leo19941227) [reply docs/source/tutorial/upstream_collection.rst:1038]: Oh I think adding the 3rd iteration of HuBERT is not necessary. I was just curious and making sure that the note is correct. I have no further issue then. (You can still submit the 3rd iteration HuBERT if you want)

**Comment by leo19941227** [comment]
Hi @ftshijt ,

I am good with the current changes and I am ready to merge this.
Let me know if you are ready or you plan to submit the 3rd iteration HuBERT in this PR.

**Comment by ftshijt** [comment]
> Hi @ftshijt ,
> 
> I am good with the current changes and I am ready to merge this. Let me know if you are ready or you plan to submit the 3rd iteration HuBERT in this PR.

Thanks again for the review. I would prefer to merge this PR. For the 3rd iteration hubert, I will commit that to the huggingface repo directly for reference purposes.

**Comment by leo19941227** [comment]
Sounds good!

**Pull request merged**